core/src/player.rs

```diff
@@ -735,20 +735,14 @@ impl Player {
         let mut display_object = run_mouse_pick(context, false).map(|obj| obj.as_displayobject());
         loop {
             if let Some(disp_obj) = display_object {
-                if disp_obj.hit_test_shape(
-                    context,
-                    *context.mouse_position,
-                    HitTestOptions::MOUSE_PICK,
-                ) {
-                    if let Some(Value::Object(obj)) = Some(disp_obj).map(|obj| obj.object()) {
-                        let mut activation = Activation::from_stub(
-                            context.reborrow(),
-                            ActivationIdentifier::root("[ContextMenu]"),
-                        );
+                if let Some(Value::Object(obj)) = Some(disp_obj).map(|obj| obj.object()) {
```

Comment:
Can this line just be
```
                if let Some(Value::Object(obj)) = disp_obj.object() {
```
?

Reply:
Fixed that, also refactored the rest of the function.